### PR TITLE
Harmeetkaur

### PR DESCRIPTION
**Pull request by Harmeetkaur13**
Fixed issue #ticket_number 307: Added duplicate brand tag check with error message

### CODE REVIEW

File: brand/admin.py [hunk 1]
```diff
@@ -326,7 +326,16 @@ def save_model(self, request, obj, form, change):
         This function is to create object of commentary model when default values are
         provided by user in admin portal and save the data in respective database.
         """
-        super().save_model(request, obj, form, change)
+        # Check if there's already a brand with the same tag
+        if (
+            Brand.objects.filter(tag=obj.tag).exists() and not change
```
Comment (RogerTangos) [inline brand/admin.py:331]: @Harmeetkaur13 - this is a picky thing, but you probably want to test for `not change` before testing for `.exists()`. The reason is that `.exists()` will have to make a database query whereas change is arelady stored in memory.

**Review by nikita-pardeshi-github — CHANGES_REQUESTED** [review]
Hi @Harmeetkaur13 This change does not work. User still get's 500 error. Basically, we need to raise validation error when user try to save the model with duplicate tag. So we want to add validation error for tag. You can update validation_tag method in Brand model like below. 

````
def validate_tag(value):
    """This is the function that is used to validate the TAG"""
    if re.match("^[A-Za-z0-9_-]*$", str(value)):
        if Brand.objects.filter(tag=value).exists():
            raise ValidationError(
                f"A brand with tag '{value}' already exists. Please edit that brand instead."
            )
        else:
            return value          
    else:
        raise ValidationError(
            "Tag can contain only alpha-numeric characters, underscores and dashes"
        )
````

**Comment by RogerTangos** [comment]
Hey @Harmeetkaur13 - this is better, but you've got to run tests before pushing. In this case, it seems like the tests will need to be updated

**Review by RogerTangos — CHANGES_REQUESTED** [review]
Please run and update tests before opening the PR

**Review by RogerTangos — APPROVED** [review]
self approving here. I've tested

**Pull request merged**